### PR TITLE
[analyzer] Split alpha core Identical Expression tests

### PR DESCRIPTION
**Pull request by vabridgers**
Split the remnant test from PR #114715, "Remove alpha.core.IdenticalExpr Checker" into seperate tests for misc-redundant-expression and bugprone-branch-clone per review comment requests.

### CODE REVIEW

File: clang-tools-extra/test/clang-tidy/checkers/bugprone/bugprone-branch-clone.cpp [hunk 1]
```diff
@@ -0,0 +1,772 @@
...
+  int* f = &k;
+  int* g = &k;
+  return (f != g); // no warning
+}
+
+int checkNotEqualCastIntPointerDeclCompare11(void) {
+  int k = 7;
+  int* f = &k;
+  return ((int*)f != (int*)f);
+// CHECK-MESSAGES-IDENTEXPR: :[[@LINE-1]]:19: warning: both sides of operator are equivalent [misc-redundant-expression]
```
Comment (5chmidti) [inline clang-tools-extra/test/clang-tidy/checkers/bugprone/bugprone-branch-clone.cpp:45]: You left-in `IDENTEXPR`-suffix checks, which are confusing when this file is not for testing the redundant expression check.

**Review by 5chmidti — COMMENTED** [review]
Please change the `RUN` lines to use `check_clang_tidy` like so:
`// RUN: %check_clang_tidy %s bugprone-branch-clone %t`, and drop the `BUGPRONEBRANCH` suffix in `bugprone-branch-clone.cpp`, as it is redundant after removing the accidental copies of `-IDENTEXPR`s.

Because the other test already had the suffix, I'd just keep it. You can add a suffix to `check_clang_tidy` via `-check-suffixes=IDENTEXPR`

**Comment by 5chmidti** [comment]
The file-names should follow the naming convention of the other tests: `checkers/misc/redundant-expression.cpp` (applies to both). 

@HerrCai0907  @PiotrZSL IMO merging these tests with the existing ones would result in too-large files, so the files would probably be named something like `redundant-expression-2.cpp`. WDYT?

**Comment by vabridgers** [comment]
I speculatively updated the filenames per @5chmidti comments. Thanks.

**Review by 5chmidti — APPROVED** [review]
Thanks.

LGTM from my side, but give it a few days in case the others want to review as well

**Comment by vabridgers** [comment]
No problem. Thanks @5chmidti !

**Pull request merged**